Voll-Text-Suche hängt sich auf

Es geschieht ab und zu, dass sich die Voll-Text-Suche mit dem folgenden Error aufhängt:

```
@400000005df8dbe5173736e4 ! @7e7g6e6i2 - Internal server error, for (GET) [/de/fulltext?id=80dfeafcda&key=SAPP] ->
@400000005df8dbe517373acc~~
@400000005df8dbe517373acc play.api.UnexpectedException: Unexpected exception[NullPointerException: null]
@400000005df8dbe517373acc >-at play.api.http.HttpErrorHandlerExceptions$.throwableToUsefulException(HttpErrorHandler.scala:276)
@400000005df8dbe517373eb4 >-at play.api.http.DefaultHttpErrorHandler.onServerError(HttpErrorHandler.scala:206)
@400000005df8dbe5173761dc >-at play.api.GlobalSettings$class.onError(GlobalSettings.scala:160)
@400000005df8dbe5173761dc >-at play.api.DefaultGlobal$.onError(GlobalSettings.scala:188)
@400000005df8dbe5173761dc >-at play.api.http.GlobalSettingsHttpErrorHandler.onServerError(HttpErrorHandler.scala:98)
@400000005df8dbe5173765c4 >-at play.core.server.netty.PlayRequestHandler$$anonfun$2$$anonfun$apply$1.applyOrElse(PlayRequestHandler.scala:100)
@400000005df8dbe517376d94 >-at play.core.server.netty.PlayRequestHandler$$anonfun$2$$anonfun$apply$1.applyOrElse(PlayRequestHandler.scala:99)
@400000005df8dbe517376d94 >-at scala.concurrent.Future$$anonfun$recoverWith$1.apply(Future.scala:344)
@400000005df8dbe51737717c >-at scala.concurrent.Future$$anonfun$recoverWith$1.apply(Future.scala:343)
@400000005df8dbe51737717c >-at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:32)
@400000005df8dbe51737811c Caused by: java.lang.NullPointerException: null
@400000005df8dbe51737811c >-at controllers.MainController.showFullTextSearchResult(MainController.java:262)
@400000005df8dbe517378504 >-at router.Routes$$anonfun$routes$1$$anonfun$applyOrElse$18$$anonfun$apply$18.apply(Routes.scala:593)
@400000005df8dbe517378504 >-at router.Routes$$anonfun$routes$1$$anonfun$applyOrElse$18$$anonfun$apply$18.apply(Routes.scala:593)
@400000005df8dbe517378cd4 >-at play.core.routing.HandlerInvokerFactory$$anon$4.resultCall(HandlerInvoker.scala:157)
@400000005df8dbe517378cd4 >-at play.core.routing.HandlerInvokerFactory$$anon$4.resultCall(HandlerInvoker.scala:156)
@400000005df8dbe5173790bc >-at play.core.routing.HandlerInvokerFactory$JavaActionInvokerFactory$$anon$14$$anon$3$$anon$1.invocation(HandlerInvoker.scala:136)
@400000005df8dbe5173794a4 >-at play.core.j.JavaAction$$anon$1.call(JavaAction.scala:73)
@400000005df8dbe5173794a4 >-at actions.MyActionCreator$1.call(MyActionCreator.java:27)
@400000005df8dbe51737988c >-at play.core.j.JavaAction$$anonfun$7.apply(JavaAction.scala:108)
@400000005df8dbe51737988c >-at play.core.j.JavaAction$$anonfun$7.apply(JavaAction.scala:108)
@400000005df92f6a36230434 [^[[31merror^[[0m] application -~
@400000005df92f6a3623081c~
```

oder 

```
@400000005df9a8332ecad7fc ! @7e7kh012h - Internal server error, for (GET) [/de/fulltext?id=2861065c07&key=sappinfo] ->
@400000005df9a8332ecadbe4~~
@400000005df9a8332ecadbe4 play.api.UnexpectedException: Unexpected exception[NullPointerException: null]
@400000005df9a8332ecadbe4 >-at play.api.http.HttpErrorHandlerExceptions$.throwableToUsefulException(HttpErrorHandler.scala:276)
@400000005df9a8332ecadfcc >-at play.api.http.DefaultHttpErrorHandler.onServerError(HttpErrorHandler.scala:206)
@400000005df9a8332ecb02f4 >-at play.api.GlobalSettings$class.onError(GlobalSettings.scala:160)
@400000005df9a8332ecb02f4 >-at play.api.DefaultGlobal$.onError(GlobalSettings.scala:188)
@400000005df9a8332ecb06dc >-at play.api.http.GlobalSettingsHttpErrorHandler.onServerError(HttpErrorHandler.scala:98)
@400000005df9a8332ecb06dc >-at play.core.server.netty.PlayRequestHandler$$anonfun$2$$anonfun$apply$1.applyOrElse(PlayRequestHandler.scala:100)
@400000005df9a8332ecb0eac >-at play.core.server.netty.PlayRequestHandler$$anonfun$2$$anonfun$apply$1.applyOrElse(PlayRequestHandler.scala:99)
@400000005df9a8332ecb0eac >-at scala.concurrent.Future$$anonfun$recoverWith$1.apply(Future.scala:344)
@400000005df9a8332ecb1294 >-at scala.concurrent.Future$$anonfun$recoverWith$1.apply(Future.scala:343)
@400000005df9a8332ecb1294 >-at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:32)
@400000005df9a8332ecb1e4c Caused by: java.lang.NullPointerException: null
@400000005df9a8332ecb2234 >-at controllers.MainController.showFullTextSearchResult(MainController.java:262)
@400000005df9a8332ecb2234 >-at router.Routes$$anonfun$routes$1$$anonfun$applyOrElse$18$$anonfun$apply$18.apply(Routes.scala:593)
@400000005df9a8332ecb261c >-at router.Routes$$anonfun$routes$1$$anonfun$applyOrElse$18$$anonfun$apply$18.apply(Routes.scala:593)
@400000005df9a8332ecb2a04 >-at play.core.routing.HandlerInvokerFactory$$anon$4.resultCall(HandlerInvoker.scala:157)
@400000005df9a8332ecb2dec >-at play.core.routing.HandlerInvokerFactory$$anon$4.resultCall(HandlerInvoker.scala:156)
@400000005df9a8332ecb2dec >-at play.core.routing.HandlerInvokerFactory$JavaActionInvokerFactory$$anon$14$$anon$3$$anon$1.invocation(HandlerInvoker.scala:136)
@400000005df9a8332ecb31d4 >-at play.core.j.JavaAction$$anon$1.call(JavaAction.scala:73)
@400000005df9a8332ecb35bc >-at actions.MyActionCreator$1.call(MyActionCreator.java:27)
@400000005df9a8332ecb35bc >-at play.core.j.JavaAction$$anonfun$7.apply(JavaAction.scala:108)
@400000005df9a8332ecb39a4 >-at play.core.j.JavaAction$$anonfun$7.apply(JavaAction.scala:108)
@400000005df9a89d2c5aa704 [^[[31merror^[[0m] application -~
@400000005df9a89d2c5aaaec~
```

--- a/app/controllers/MainController.java
+++ b/app/controllers/MainController.java
@@ -258,6 +258,10 @@ public class MainController extends Controller {
 
             FullTextEntry entry = getFullTextEntryWithId(lang, row_id);
 
+            if (entry == null) {
+                return notFound("Result with this id is not found");
+            }
+
             // This operation takes time...
             List<Article> list_of_articles = getArticlesFromRegnrs(lang, entry.getRegnrs()).join();
 
@@ -705,16 +709,20 @@ public class MainController extends Controller {
     public FullTextEntry getFullTextEntryWithId(String lang, String rowId) {
         try {
             Connection conn = lang.equals("de") ? frequency_de_db.getConnection() : frequency_fr_db.getConnection();
-            Statement stat = conn.createStatement();
-            String query = "select * from " + FREQUENCY_TABLE + " where id='" + rowId + "'";
+            String query = "select * from " + FREQUENCY_TABLE + " where id = ?";
+            PreparedStatement stat = conn.prepareStatement(query);
+            stat.setString(1, rowId);
 
-            ResultSet rs = stat.executeQuery(query);
+            ResultSet rs = stat.executeQuery();
+            if (!rs.next()) {
+                return null;
+            }
             FullTextEntry full_text_entry = cursorToFullTextEntry(rs);
             conn.close();
             if (full_text_entry!=null)
                 return full_text_entry;
         } catch (SQLException e) {
-            System.err.println(">> Frequency DB: SQLException in getFullTextEntryWithId");
+            System.err.println(">> Frequency DB: SQLException in getFullTextEntryWithId: " + e.getMessage());
         }
         return null;
     }

